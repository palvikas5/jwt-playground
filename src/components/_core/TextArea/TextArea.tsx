import React from 'react';
import { StyledTextArea } from './textArea.style';
import { noop } from '../../../utils/noop';

export interface TextAreaProps {
  value: string;
  'aria-label': string;
  placeholder?: string;
  onChange?: (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
  noResize?: boolean;
}

const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
  (
    {
      value,
      onChange,
      placeholder,
      noResize,
      'aria-label': ariaLabel,
    }: TextAreaProps,
    ref,
  ) => {
    return (
      <StyledTextArea
        noResize={noResize}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        aria-label={ariaLabel}
        ref={ref}
      />
    );
  },
);

TextArea.defaultProps = {
  onChange: noop,
};

export default TextArea;
