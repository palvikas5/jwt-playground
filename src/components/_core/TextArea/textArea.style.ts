/* stylelint-disable value-keyword-case */

import styled from 'styled-components';
import { TextAreaProps } from './TextArea';

const StyledTextArea = styled.textarea<TextAreaProps>`
  display: block;
  width: 100%;
  height: inherit;
  flex: 1;
  padding: 10px;
  font-size: 1.7rem;
  border-radius: 3px;
  border: 1px solid #2f3b52;
  outline: 0;
  background-color: #20293c;
  color: white;
  box-shadow: 0 0 3px 1px rgb(0 0 0 / 0.16);
  ${({ noResize }) => noResize && `resize: none;`}

  &:focus {
    border-color: #3e8efa;
  }
`;

export { StyledTextArea };
