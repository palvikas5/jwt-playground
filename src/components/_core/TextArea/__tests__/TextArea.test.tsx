import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import TextArea from '../TextArea';

describe('<TextArea />', () => {
  it('should render successfully', () => {
    expect.assertions(1);

    const { container } = render(
      <TextArea
        value='textarea value'
        placeholder='enter value here'
        aria-label='aria text'
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should contains value as children', () => {
    expect.assertions(1);

    const { getByText } = render(
      <TextArea value='textarea value' aria-label='aria text' />,
    );
    expect(getByText(/^textarea value$/)).toBeInTheDocument();
  });

  it('should render successfully without resize controls', () => {
    expect.assertions(1);

    const { container } = render(
      <TextArea
        value='textarea value'
        placeholder='enter value here'
        aria-label='aria text'
        noResize
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should trigger onChange event', () => {
    expect.assertions(2);

    const mockChangeHandler = jest.fn();
    const { getByText } = render(
      <TextArea
        value='textarea value'
        onChange={mockChangeHandler}
        aria-label='aria text'
      />,
    );
    const textarea = getByText(/^textarea value$/);
    expect(textarea.textContent).toBe('textarea value');
    fireEvent.change(textarea, { target: { value: 'changed-value' } });
    expect(mockChangeHandler).toHaveBeenCalledTimes(1);
  });
});
