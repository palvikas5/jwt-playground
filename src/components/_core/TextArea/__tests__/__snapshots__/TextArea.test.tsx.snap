// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> should render successfully 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: inherit;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 10px;
  font-size: 1.7rem;
  border-radius: 3px;
  border: 1px solid #2f3b52;
  outline: 0;
  background-color: #20293c;
  color: white;
  box-shadow: 0 0 3px 1px rgb(0 0 0 / 0.16);
}

.c0:focus {
  border-color: #3e8efa;
}

<div>
  <textarea
    aria-label="aria text"
    class="c0"
    placeholder="enter value here"
  >
    textarea value
  </textarea>
</div>
`;

exports[`<TextArea /> should render successfully without resize controls 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: inherit;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 10px;
  font-size: 1.7rem;
  border-radius: 3px;
  border: 1px solid #2f3b52;
  outline: 0;
  background-color: #20293c;
  color: white;
  box-shadow: 0 0 3px 1px rgb(0 0 0 / 0.16);
  resize: none;
}

.c0:focus {
  border-color: #3e8efa;
}

<div>
  <textarea
    aria-label="aria text"
    class="c0"
    placeholder="enter value here"
  >
    textarea value
  </textarea>
</div>
`;
